points/forwarder.go

```diff
@@ -0,0 +1,130 @@
+package points
+
+import (
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rcrowley/go-metrics"
+	"github.com/wavefronthq/go-proxy/api"
+)
+
+// Interface that forwards points to a Wavefront instance.
+type PointForwarder interface {
+	init()
+	addPoint(point string)
+	incrementBlockedPoint()
+	stop()
+}
+
+type DefaultPointForwarder struct {
+	name            string
+	prefix          string
+	workUnitId      string
+	dataFormat      string
+	points          []string
+	maxBufferSize   int
+	maxFlushSize    int
+	mtx             sync.Mutex
+	api             api.WavefrontAPI
+	pushTicker      *time.Ticker
+	pointsReceived  metrics.Counter
+	pointsBlocked   metrics.Counter
+	pointsQueued    metrics.Counter
+	pointsSent      metrics.Counter
+	pointsFlushTime metrics.Timer
+}
+
+func (f *DefaultPointForwarder) init() {
+	f.pointsReceived = metrics.GetOrRegisterCounter("points."+f.prefix+".received", nil)
+	f.pointsBlocked = metrics.GetOrRegisterCounter("points."+f.prefix+".blocked", nil)
+	f.pointsQueued = metrics.GetOrRegisterCounter("points."+f.prefix+".queued", nil)
+	f.pointsSent = metrics.GetOrRegisterCounter("points."+f.prefix+".sent", nil)
+	f.pointsFlushTime = metrics.GetOrRegisterTimer("push."+f.prefix+".duration", nil)
+	go f.flushPoints()
+}
+
+func (f *DefaultPointForwarder) flushPoints() {
+	for range f.pushTicker.C {
+		f.pointsFlushTime.Time(func() {
+			f.post(f.getPointsBatch())
+		})
+	}
+	log.Printf("%s: exiting flushPoints", f.name)
+}
+
+func (f *DefaultPointForwarder) stop() {
+	f.pushTicker.Stop()
+}
+
+func min(x, y int) int {
```

Comment:
math.min or something?

Reply:
Strangely, the in-built Math.min (https://golang.org/pkg/math/#Min) only takes floats. It is apparently idiomatic in Go to just write a simple min function for other cases.